llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -185,57 +193,116 @@ def validate_interval(interval: Union[str, int, Time],
     return result
 
 
+def validate_eval_run_config(
+        eval_run_config: Optional[Dict[str, Any]]) -> Dict[str, Any]:
+
+    if isinstance(eval_run_config, DictConfig):
+        parsed_run_config = om.to_container(eval_run_config)
+        run_config = cast(Dict[str, Any], parsed_run_config)
+    elif eval_run_config is None:
+        return {}
+    else:
+        run_config = eval_run_config.copy()
+
+    if not run_config:
+        return {}
+
+    supported_keys = {'image', 'command', 'compute', 'scheduling'}
+    for key in run_config:
+        if key not in supported_keys:
+            raise ValueError(
+                f'Unsupported eval run config key {key}. Supported keys: {supported_keys}'
+            )
+
+    for dict_field in ('scheduling', 'compute'):
```

Comment:
Is there a simpler way to do this validation using anything from msdk? The very specific type checking feels a little rough. Was wondering if maybe we can just construct the config objects, or call some validate function from msdk.